tests/rptest/tests/delete_records_test.py

```diff
@@ -775,3 +783,153 @@ class PartitionMoveExceptionReporter:
+    """
+    def __init__(self, test_context):
+        super().__init__(test_context=test_context, num_brokers=3)
+        self.ctx = test_context
+        self.rpk = RpkTool(self.redpanda)
+        self.admin = Admin(self.redpanda)
+        self.client = DefaultClient(self.redpanda)
+
+    def get_topic_info(self, topic_name):
+        topics_info = list(self.rpk.describe_topic(topic_name))
+        self.logger.info(topics_info)
+        assert len(topics_info) == 1
+        return topics_info[0]
+
+    def make_delete_records_for_cli(self,
+                                    topic_partition_offsets: list[tuple[str,
+                                                                        int,
+                                                                        int]]):
+        delete_records_json = {"version": 1, "partitions": []}
+        for t, p, o in topic_partition_offsets:
+            delete_records_json["partitions"].append({
+                "topic": t,
+                "partition": p,
+                "offset": o
+            })
+        return delete_records_json
+
+    def execute_delete_records(self, delete_records_json: dict):
+        kafka_tools = KafkaCliTools(self.redpanda)
+        res = kafka_tools.delete_records(delete_records_json)
+        split_str = res.split('\n')
+        assert split_str[-1] == ""
+        partition_watermark_lines = split_str[2:-1]
+        result_list: list[RpkTrimOffsetResponse] = []
+        for partition_watermark_line in partition_watermark_lines:
+            topic_partition_str, result_str = partition_watermark_line.strip(
+            ).split('\t')
+            topic_partition_str_split = topic_partition_str.split(
+                "partition: ")[1]
+            topic_partition_split_index = topic_partition_str_split.rfind('-')
+            topic_str = topic_partition_str_split[:topic_partition_split_index]
+            partition_str = topic_partition_str_split[
+                topic_partition_split_index:]
+            new_start_offset = -1
+            error_msg = ""
+            if "error" in result_str:
+                error_msg = result_str.split("error: ")[1]
+            elif "low_watermark" in result_str:
+                new_start_offset = int(result_str.split("low_watermark: ")[1])
+            result_list.append(
+                RpkTrimOffsetResponse(topic_str, int(partition_str),
+                                      new_start_offset, error_msg))
+        return result_list
+
+    @cluster(num_nodes=3)
+    def test_delete_records_empty_topic(self):
+        empty_topic = TopicSpec(name=TEST_TOPIC_NAME,
+                                partition_count=1,
+                                replication_factor=3)
+        self.client.create_topic(empty_topic)
+        delete_records_tpos = [(TEST_TOPIC_NAME, 0, 0)]
+        delete_records_json = self.make_delete_records_for_cli(
+            delete_records_tpos)
+
+        delete_records_result = self.execute_delete_records(
```

Comment:
make/execute are always called together. just make it one (wrapper) method

Reply:
this will make very easy to understand what the test is actually testing

make non-empty error_msg throw in that helper method or in execute_delete_records so you don't have to assert it is empty everywhere

where you need to assert that it throws, test it with `rptest.util.expect_exception`

Reply:
Refactored `execute_kafka_delete_records()` to be the one wrapper method needed to make a request, and further cleaned up test.